File: src/main/java/com/example/ajaxonboardingproject/service/mapper/CinemaHallMapper.java
package com.example.ajaxonboardingproject.service.mapper;

import com.example.ajaxonboardingproject.dto.request.CinemaHallRequestDto;
import com.example.ajaxonboardingproject.dto.response.CinemaHallResponseDto;
import com.example.ajaxonboardingproject.model.CinemaHall;
import org.springframework.stereotype.Component;

@Component
public class CinemaHallMapper implements RequestDtoMapper<CinemaHallRequestDto, CinemaHall>,
        ResponseDtoMapper<CinemaHallResponseDto, CinemaHall> {
    @Override
    public CinemaHall mapToModel(CinemaHallRequestDto dto) {
        CinemaHall cinemaHall = new CinemaHall();
        cinemaHall.setDescription(dto.getDescription());
        cinemaHall.setCapacity(dto.getCapacity());
        return cinemaHall;
    }

    @Override
    public CinemaHallResponseDto mapToDto(CinemaHall cinemaHall) {
        return new CinemaHallResponseDto(
                cinemaHall.getId(),
                cinemaHall.getCapacity(),
                cinemaHall.getDescription());
    }
}

File: src/main/java/com/example/ajaxonboardingproject/service/mapper/CinemaHallMapper.kt
package com.example.ajaxonboardingproject.service.mapper

import com.example.ajaxonboardingproject.dto.request.CinemaHallRequestDto
import com.example.ajaxonboardingproject.dto.response.CinemaHallResponseDto
import com.example.ajaxonboardingproject.model.CinemaHall
import org.springframework.stereotype.Component

fun CinemaHallMapper.mapToModel(dto : CinemaHallRequestDto) : CinemaHall {
    return CinemaHall(
            description = dto.description,
            capacity = dto.capacity
    )
}

fun CinemaHallMapper.mapToDto(model: CinemaHall): CinemaHallResponseDto {
    return model.id?.let { CinemaHallResponseDto(
            id = it,
            capacity = model.capacity,
            description = model.description)}!!
}

@Component
class CinemaHallMapper
